src/instructlab/config/init.py

```diff
@@ -357,3 +439,46 @@ def get_separator(text: str) -> str:
         terminal_width = text_length
     separator_length = min(text_length, terminal_width)
     return "-" * separator_length
+
+
+def get_chip_name() -> tuple[str, bool, bool]:
+    """
+    get_chip_name returns the name of the processor on the system (Linux or Mac for now), whether the system is linux, and whether the chip is a CPU or not.
+    """
+    # Standard
+    import platform
+    import subprocess
+
+    system = platform.system()
+
+    if system == "Darwin":  # macOS
+        try:
+            # macOS: Use system_profiler for detailed hardware info
+            result = subprocess.run(
+                ["system_profiler", "SPHardwareDataType"],
+                capture_output=True,
+                text=True,
+                check=True,
+            )
+            for line in result.stdout.splitlines():
+                if "Chip" in line or "Processor Name" in line:
+                    return line.split(":")[-1].strip().lower(), False, True
+        except subprocess.CalledProcessError:
+            return "Unsupported", False, True
+
+    elif system == "Linux":  # Linux
+        try:
+            # Linux: Read /proc/cpuinfo for CPU model information
+            with open("/proc/cpuinfo", "r", encoding="utf-8") as f:
+                for line in f:
+                    if "model name" in line:
+                        # print(line)
+                        return (
+                            line.split(":")[-1].strip().lower().split(" ")[0],
+                            True,
+                            True,
+                        )
+        except FileNotFoundError:
+            return "Unsupported", True, True
+
```

Comment:
can we add an else to inform users we only support linux/mac?

Reply:
sure! I can add a log warning